content_server.py

```diff
@@ -1,83 +1,83 @@
+import urlparse
 from nose.tools import set_trace
 from core.config import Configuration
 from core.model import (
+    Collection,
     DataSource,
     Identifier,
 )
 from core.opds_import import (
     SimplifiedOPDSLookup,
     OPDSImporter,
 )
-from core.coverage import (
-    CoverageProvider,
-    CoverageFailure,
-)
+from core.coverage import CatalogCoverageProvider
 from core.util.http import BadResponseException
 
-class ContentServerException(Exception):
-    # Raised when the ContentServer can't connect or returns bad data
-    pass
 
-class ContentServerCoverageProvider(CoverageProvider):
-    """Checks the OA Content Server for metadata about Gutenberg books
-    and books identified by URI.
+class LookupClientCoverageProvider(CatalogCoverageProvider):
+    """Uses the Library Simplified OPDS Lookup Protocol to get
+    extra information about books in a Catalog.
     """
 
-    CONTENT_SERVER_RETURNED_WRONG_CONTENT_TYPE = "Content Server served unhandleable media type: %s"
-
-    def __init__(self, _db, content_server=None):
-        self._db = _db
-        if not content_server:
-            content_server_url = Configuration.integration_url(
-                Configuration.CONTENT_SERVER_INTEGRATION, required=True)
-            content_server = SimplifiedOPDSLookup(content_server_url)
-        self.content_server = content_server
-        self.importer = OPDSImporter(self._db, DataSource.OA_CONTENT_SERVER)
-        input_identifier_types = [Identifier.GUTENBERG_ID, Identifier.URI]
-        output_source = DataSource.lookup(
-            self._db, DataSource.OA_CONTENT_SERVER
-        )
-        super(ContentServerCoverageProvider, self).__init__(
-                "OA Content Server Coverage Provider",
-                input_identifier_types, output_source, batch_size=10
+    # TODO: We should rename this because in theory it can be used
+    # other places, but in practice this is it.
+    SERVICE_NAME = "OA Content Server Coverage Provider"
+    PROTOCOL = Collection.OPDS_IMPORT
+    
+    OPDS_SERVER_RETURNED_WRONG_CONTENT_TYPE = "OPDS Server served unhandleable media type: %s"
+   
+    def __init__(self, collection, **kwargs):
+        self.DATA_SOURCE_NAME = collection.data_source.name
+        super(LookupClientCoverageProvider, self).__init__(
+            collection, **kwargs
         )
 
+        # Assume that this collection's OPDS server also implements
+        # the lookup protocol.
+        feed_url = collection.external_account_id
+        root = urlparse.urljoin(feed_url, '/')
+        self.lookup_client = self._lookup_client(root)
+        self.importer = self._importer()
```

Comment:
This lets us take the 'content server' integration out of the JSON config. We now perform an OPDS lookup on the server that identifies the Collection (which, in general, will be the content server).